src/Umbraco.PublishedCache.HybridCache/Persistence/DatabaseCacheRepository.cs

```diff
@@ -429,32 +429,46 @@ private void RebuildMemberDbCache(IContentCacheDataSerializer serializer, int gr
             IEnumerable<IMember> descendants =
                 _memberRepository.GetPage(query, pageIndex++, groupSize, out total, null, Ordering.By("Path"));
             ContentNuDto[] items = descendants.Select(m => GetDtoFromContent(m, false, serializer)).ToArray();
-            Database.BulkInsertRecords(items);
+            _ = Database.BulkInsertRecords(items);
             processed += items.Length;
         }
         while (processed < total);
     }
 
-    private void DeleteForObjectType(Guid nodeObjectType) =>
-        Database.Execute(
-            @"
-                DELETE FROM cmsContentNu
-                WHERE cmsContentNu.nodeId IN (
-                    SELECT id FROM umbracoNode WHERE umbracoNode.nodeObjectType = @objType
-                )",
-            new { objType = nodeObjectType });
-
-    private void DeleteForObjectTypeAndContentTypes(Guid nodeObjectType, IReadOnlyCollection<int> contentTypeIds) =>
-        Database.Execute(
-            $@"
-                DELETE FROM cmsContentNu
-                WHERE cmsContentNu.nodeId IN (
-                    SELECT id FROM umbracoNode
-                    JOIN {Constants.DatabaseSchema.Tables.Content} ON {Constants.DatabaseSchema.Tables.Content}.nodeId=umbracoNode.id
-                    WHERE umbracoNode.nodeObjectType = @objType
-                    AND {Constants.DatabaseSchema.Tables.Content}.contentTypeId IN (@ctypes)
-                )",
-            new { objType = nodeObjectType, ctypes = contentTypeIds });
+    private void RemoveByObjectType(Guid objectType, IReadOnlyCollection<int>? contentTypeIds)
+    {
+        // remove all - if anything fails the transaction will rollback
+        Sql<ISqlContext> sql;
+        if (contentTypeIds is null || contentTypeIds.Count == 0)
+        {
+            sql = Sql()
+                .Delete<ContentNuDto>()
+                .WhereIn<ContentNuDto>(
+                    x => x.NodeId,
+                    Sql().Select<NodeDto>(n => n.NodeId)
+                        .From<NodeDto>()
+                        .Where<NodeDto>(n => n.NodeObjectType == objectType));
+            _ = Database.Execute(sql);
+        }
+        else
+        {
+            // assume number of ctypes won't blow IN(...)
+            // must support SQL-CE
+            sql = Sql()
+                .Delete<ContentNuDto>()
+                .WhereIn<ContentNuDto>(
+                    x => x.NodeId,
+                    Sql()
+                    .Select<NodeDto>(n => n.NodeId)
+                    .From<NodeDto>()
+                    .InnerJoin<ContentDto>()
+                    .On<NodeDto, ContentDto>((n, c) => n.NodeId == c.NodeId)
+                    .Where<NodeDto, ContentDto>((n, c) =>
+                        n.NodeObjectType == objectType
+                        && contentTypeIds.Contains(c.ContentTypeId)));
```

Comment:
```suggestion
                        n.NodeObjectType == objectType &&
                        contentTypeIds.Contains(c.ContentTypeId)));
```

Nit-pick: I just more typically see code with && and || laid out this way.

Reply:
Fine.